aravisApp/src/ADAravis.cpp

```diff
@@ -801,7 +806,13 @@ asynStatus ADAravis::processBuffer(ArvBuffer *buffer) {
 
     /* Put the frame number and time stamp into the buffer */
     pRaw->uniqueId = imageCounter;
-    pRaw->timeStamp = arv_buffer_get_timestamp(buffer) / 1.e9;
+    int timestampMode;
+    getIntegerParam(AravisTimestampMode, &timestampMode);
+    if (timestampMode) {
+        pRaw->timeStamp = arv_buffer_get_system_timestamp(buffer) / 1.e9;
```

Comment:
Please can you get it from epicsTS as per ADVimba?
https://github.com/areaDetector/ADVimba/blob/master/vimbaApp/src/ADVimba.cpp#L525-L534

You will need to move `updateTimeStamp` above this block so it looks the same as ADVimba

Reply:
The problem is that I want that timestamp because it's more precise, this is saved on received of the first packet by the aravis library, while if we saved the epicsTS, that is saved after receiving all the thousands of packets for the full frame, therefore it's less deterministic... should I add a third option? so we can choose between Camera, EPICS and System?

Reply:
Hmm, I'm not sure. I think we could either:
1. Have 3 modes, Camera, EPICS, System. epicsTS is always taken from time buffer is completed, timeStamp is either from camera, when first packet received, or when buffer is completed
2. Have 2 modes, Camera or EPICS. epicsTS is always taken from time first packet received, timeStamp is either from camera or when first packet received

If we want 2. then I guess we should make an epicsTimeStamp then call [setTimeStamp](https://github.com/epics-modules/asyn/blob/251fa468573f3c779de69706d84998d0d5f31aa5/asyn/asynPortDriver/asynPortDriver.cpp#L3575-L3578) instead of updateTimeStamp as present?

@MarkRivers do you have a preference?

Reply:
Seeing @MarkRivers comment https://github.com/areaDetector/ADAndor/pull/58#issuecomment-1815231345 I guess 1. is the better solution here